presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -96,4 +99,23 @@ public static Slice charToVarcharCast(@LiteralParameter("y") Long y, @SqlType("c
 
         return buffer;
     }
+
+    @ScalarOperator(OperatorType.SATURATED_FLOOR_CAST)
+    @SqlType("char(y)")
+    @LiteralParameters({"x", "y"})
+    public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") Long y, @SqlType("varchar(x)") Slice slice)
```

Comment:
Add comment that it does not actually return greatest value in CHAR(Y) smaller that given value from VARCHAR(Y). And that it is not a problem.